### DISCUSSION TITLE
Can we run experiments on a particular model with providing CSV file

### DISCUSSION THREAD

**Discussion by alihassan90** [pybamm-team/PyBaMM#3000] [General]
In one of the notebooks in **"getting started"** section on the PyBamm website the experiments is written like this:

 _("Discharge at C/10 for 10 hours or until 3.3 V",
        "Rest for 1 hour",
        "Charge at 1 A until 4.1 V",
        "Hold at 4.1 V until 50 mA",
        "Rest for 1 hour")_,
        
**Question 1:**
 Is there a way I can give my model and experiment values through a CSV or Excel file? or does it have to be in the same format (wordings only). 

**Question 2:** 
Can I use PyBAMm in a closed loop, i.e. generate an experiment (different current/voltage values) for different cycles and feed to PyBAMm and see how the battery model degrades?

**Comment by brosaplanella** [comment]
**Question 1:** you cannot directly read experiments from csv, but you can use pandas to read the csv file and then process it manually to convert it to the right format.

**Question 2:** I don't think I understand your question. What do you exactly mean by closed loop?

**Reply by alihassan90** [reply]
Thank you Dr. Planella for the replies. By closed loop I meant that lets say I want to optimize battery life time as a trade off between its charge-discharge pattern for a grid storage. So I run my optimization problem in Pyomo and integrate that with PyBAMm and see how battery degrades over time. This is like instead of running optimization problem separately and feed the charge-discharge data to run PyBAMm model I integrate both some how so PyBAMm comes in a loop somehow. I know it is just an idea and may not be possible to do so.

**Reply by brosaplanella** [reply]
It should be possible but would require a fair amount of coding. Note that you can use formatted strings to pass values on the charge and discharge steps of the experiment.
